### DISCUSSION TITLE
Pi Pico: Thonny/CTRL-C with two threads

### DISCUSSION THREAD

**Discussion by ESPWarren** [micropython/micropython#9617] [Hardware & Peripherals] [answered]
I have searched for an answer to this but could not find other reference...

I am using a Pico W. It has code to do stuff in two threads. It works fine - I can see the output of print statements from both cores.

When I click CTRL-C, the first thread obviously stops. But the second one keeps going and no amount of CTRL-C clicking can stop it!

Any ideas?

Just to say, if I disconnect the board and CTRL-C quickly enough (before the second thread is launched) then there is no probem...)

**Comment by peterhinch** [comment, marked as answer]
You could use a shared global variable, say `STOP = False`. In the main thread, trap `KeyboardInterrupt` and set `STOP = True`. On the second core, periodically check the variable. In fact it might be worth trapping any exception and setting `STOP = True`:
```python
STOP = False
try:
    run_my_application()
except:  # ctrl-c or program error
    STOP = True
```
